### DISCUSSION TITLE
[TypeScript] [button] ElButton type 可以自定义吗

### DISCUSSION THREAD

**Discussion by zhengyimeng** [element-plus/element-plus#10975] [Q&A]
Feature type: **`TypeScript`**

## Details

- Enhanced component name: `button`

## Description

我有一个次级按钮样式，但是改任何一个 type="['', success, info, warning, primary, error ..]" 的样式，都不太合适
我让他的 type=”secondary“。<el-button type="secondary"></el-button>
并且去自定义 .el-button--secondary 的样式
但是控制台会出现很多警告。【Invalid prop: validation failed for prop "type". Expected one of ["default", "primary", "success", "warning", "info", "danger", "text", ""], got value "secondary"】

### Additional comments

```
// 当前可以生效，但是 vue-tsc --noEmit 编译没法通过，而且我不希望他报警告。是否有方法可以实现？
<el-button type="secondary"></el-button>
```

**Comment by chenxch** [comment]
告警是正常的，因为有值校验，如果ts单行忽略不行的话，试试看全文ts忽略
 `// @ts-nocheck`

**Comment by zhengyimeng** [comment]
> 告警是正常的，因为有值校验，如果ts单行忽略不行的话，试试看全文ts忽略 `// @ts-nocheck`

我不想忽视告警，而是解决告警
